dataactbroker/handlers/fileHandler.py

```diff
@@ -726,9 +726,30 @@ def call_d_file_api(self, file_type_name, cgac_code, start_date, end_date, job):
                 # Check for numFound = 0
                 if "numFound='0'" in self.get_xml_response_content(api_url):
                     sess = GlobalDB.db().session
-                    # If the call to the external API wasn't successful, mark the job as failed
-                    job.error_message = "%s data unavailable for the specified date range" % file_type
-                    job.job_status_id = JOB_STATUS_DICT['failed']
+                    # No results found, skip validation and mark as finished
+                    sess.query(JobDependency). \
+                        filter(JobDependency.prerequisite_id == job.job_id). \
+                        delete(synchronize_session='fetch')
+                    mark_job_status(job.job_id, "finished")
+                    job.filename = None
+
+                    val_job = sess.query(Job).filter_by(
+                        submission_id=job.submission_id,
+                        file_type_id=FILE_TYPE_DICT[file_type_name],
+                        job_type_id=JOB_TYPE_DICT['csv_record_validation']
+                    ).one_or_none()
```

Comment:
Added an optional param to account for the detached generator's usage of the function.